staging/src/k8s.io/api/storage/v1beta1/types.go

```diff
@@ -291,10 +291,41 @@ type CSIDriverSpec struct {
 	// "csi.storage.k8s.io/pod.name": pod.Name
 	// "csi.storage.k8s.io/pod.namespace": pod.Namespace
 	// "csi.storage.k8s.io/pod.uid": string(pod.UID)
+	// "csi.storage.k8s.io/ephemeral": "true" iff the volume is an ephemeral inline volumed
+	//                                 defined by a CSIVolumeSource, otherwise "false"
 	// +optional
 	PodInfoOnMount *bool `json:"podInfoOnMount,omitempty" protobuf:"bytes,2,opt,name=podInfoOnMount"`
+
+	// If set, Mode defines how this CSI volume driver can be used.
+	// The default is "persistent", which is the usage defined by
+	// the CSI specification.
+	//
+	// Other modes are:
+	// - "ephemeral": only use this driver for inline ephemeral volumes (CSIVolumeSource)
+	// - "persistent+ephemeral": both modes supported
+	//
+	// More modes may be added in the future.
+	// +optional
+	Mode *CSIDriverMode `json:"mode,omitempty" protobuf:"bytes,3,opt,name=mode"`
 }
 
+// CSIDriverMode is an enumeration of possible usage modes of a CSI driver.
+// More modes may be added in the future.
+type CSIDriverMode string
+
+const (
+	// PersistentDriverMode explicitly confirms that the driver implements
+	// the full CSI spec. It is the default when CSIDriverSpec.Mode is not
+	// set.
+	PersistentDriverMode CSIDriverMode = "persistent"
```

Comment:
style: API constants should always be `CamelCase`, so `Persistent`, `Ephemeral`, `All`.  We generally do not use "composition constants" (`persistent+ephemeral`).

It may be more appropriate to use an array here, since if we added a third mode in the future, we don't want to have to do `AB`, `BC`, etc.  This seems like an array of supported modes, so `[Persistent,Ephemeral]`, `[Persistent]`, etc.

Reply:
The effect is that in the driver info yaml, the values also have to be capitalized. Is that really intended?
```
apiVersion: storage.k8s.io/v1beta1
kind: CSIDriver
metadata:
  name: hostpath.csi.k8s.io
spec:
  # Supports both modes, but needs pod info for that to determine the actual mode.
  podInfoOnMount: true
  volumeModes:
  - Persistent
  - Ephemeral
```

Or do I have to implement some kind of mapping?

Reply:
Looks like this is intended, see `volumeBindingMode: Immediate` (https://kubernetes.io/docs/concepts/storage/storage-classes/#the-storageclass-resource).